UIKitDZ/UIKitDZ/ViewController.swift

```diff
@@ -8,9 +8,129 @@
+        label.text = "Result: "
+        return label
+    }()
+    
+    private var hiddenNumber = 0
+    
+    private lazy var findNumberButton: UIButton = {
+        let button = UIButton(configuration: .filled())
+        button.setTitle("Find number", for: .normal)
+        button.titleLabel?.textAlignment = .center
+        button.frame = CGRect(x: 20, y: 240, width: 100, height: 50)
+        button.addTarget(self, action: #selector(findNumber(paramTarget:)), for: .touchUpInside)
+        return button
+    }()
+    
+    private lazy var findNumberTextField: UITextField = {
+        let textField = UITextField()
+        textField.placeholder = "Let's try to guess the number"
+        textField.frame = CGRect(x: 20, y: 320, width: 250, height: 40)
+        textField.isHidden = true
+        return textField
+    }()
+    
+    private lazy var tryButton: UIButton = {
+        let button =  UIButton(configuration: .filled())
+        button.setTitle("Try", for: .normal)
+        button.frame = CGRect(x: 20, y: 400, width: 100, height: 40)
+        button.isHidden = true
+        button.addTarget(self, action: #selector(tryToFindNumber(paramTarget:)), for: .touchUpInside)
+        return button
+    }()
+    
     override func viewDidLoad() {
         super.viewDidLoad()
-        // Do any additional setup after loading the view.
+        setupUI()
+    }
+    
+    override func viewDidAppear(_ animated: Bool) {
+        alertIntro()
+    }
+    
+    func setupUI() {
+        view.addSubview(nameLabel)
+        view.addSubview(additionButton)
+        view.addSubview(resultLabel)
+        view.addSubview(findNumberButton)
+        view.addSubview(findNumberTextField)
+        view.addSubview(tryButton)
+    }
+    
+    func alertIntro() {
+        let greetingAlertController = UIAlertController(title: "Hello",
+                                                        message: "Enter your name",
+                                                        preferredStyle: .alert)
+        let okAlertAction = UIAlertAction(title: "ok", style: .default) { _ in
+            self.nameLabel.text? += greetingAlertController.textFields?.first?.text ?? ""
+        }
+        greetingAlertController.addTextField()
+        greetingAlertController.addAction(okAlertAction)
+        self.present(greetingAlertController, animated: true)
+    }
+    
+    @objc func addition(paramTarget: UIButton) {
+        let additionAlertController = UIAlertController(title: "Addition",
+                                                        message: "Enter two value",
+                                                        preferredStyle: .alert)
+        additionAlertController.addTextField()
+        additionAlertController.addTextField()
+        let resultAlertAction = UIAlertAction(title: "Result", style: .default) { _ in
+            guard let firstTerm = Int(additionAlertController.textFields?[0].text ?? "") else { return }
```

Comment:
```suggestion
            guard 
               let firstTerm = Int(additionAlertController.textFields?[0].text ?? ""),
               let secondTerm = Int(additionAlertController.textFields?[1].text ?? "") 
             else {
                return 
              }
```
в один гуард  и обращение по индексу не безопасное